frontends/verilog/verilog_frontend.cc

```diff
@@ -672,6 +674,86 @@ struct VerilogDefines : public Pass {
 	}
 } VerilogDefines;
 
+struct VerilogFileList : public Pass {
+	VerilogFileList() : Pass("read_verilog_file_list", "Parse a Verilog file list") {}
+	void help() override
+	{
+		//   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+		log("\n");
+		log("    read_verilog_file_list [options]\n");
+		log("\n");
+		log("Parse a Verilog file list, and pass the list of Verilog files to read_verilog command.\n");
+		log("\n");
+		log("    -F file_list_path\n");
+		log("        File list file contains list of Verilog files to be parsed, any\n");
+		log("        ' path is treated relative to the file list file'\n");
+		log("\n");
+		log("    -f file_list_path\n");
+		log("        File list file contains list of Verilog files to be parsed, any\n");
+		log("        ' path is treated relative to current working directroy'\n");
+		log("\n");
+	}
+
+	void parse_file_list(const std::string &file_list_path, RTLIL::Design *design, bool relative_to_file_list_path)
+	{
+		std::ifstream flist(file_list_path);
+		if (!flist.is_open()) {
+			log_error("Verilog file list file does not exist");
+			exit(1);
+		}
+
+		std::filesystem::path file_list_parent_dir = std::filesystem::path(file_list_path).parent_path();
+
+		std::string v_file_name;
+		while (std::getline(flist, v_file_name)) {
+			if (v_file_name.empty()) {
+				continue;
+			}
+
+			std::string verilog_file_path;
+			if (relative_to_file_list_path) {
+				verilog_file_path = file_list_parent_dir.string() + '/' + v_file_name;
```

Comment:
If you're using `<filesystem>` you should use the `/` operator for portability.  i.e. `verilog_file = file_list_parent_dir / v_file_name`.  It would also save on the converting it back into a path alter to check the extension, and you only convert to string when you pass it to `read_verilog`.